### ISSUE TITLE
Bug: Cakewallet for Android 4.4.1 - Creating two wallets of different coins with the same name results in loss of access to first wallet

### ISSUE THREAD

**Issue by XMR-Boats** [cake-tech/cake_wallet#378]
Came across this few weeks ago when on-boarding new people into crypto currencies (yay!).

TL;DR; If you create a wallet of one coin and call it _widgets_ and then create a wallet for another coin and call it _widgets_ the stored secrets for the first wallet are overwritten or corrupted and access to the wallet is forever lost. Wallet needs to be restored from seeds which means loss of meta data. 

The long story:

**Tested on**
- Cakewallet for Android 4.4.1 from Google Play store on a Samsung device running Android 12
- Cakewallet for Android 4.4.1 from Github APK on a OnePlus 6T running Android 11
![aopfew3](https://user-images.githubusercontent.com/107238028/173058301-6dea9207-c321-4569-94c4-a37ff49ce18f.jpeg)

**Steps to reproduce**

1) Download and install Cakewallet
2) Create LTC wallet. Name it _widgets_
3) Assign a PIN to it, enable biometrics if desired. It has no bearing on the outcome.
4) Confirm wallet is working.
5) Close and reopen Cakewallet. Confirm wallet is still working and unlocks with the PIN or biometrics as normal.
6) Create a XMR wallet. Name it _widgets_, too.
~~7) Assign a PIN to it, enable biometrics if desired.~~
8) Switch back to the first wallet, the LTC wallet. 
![dpfg3555555555](https://user-images.githubusercontent.com/107238028/173058441-f9988d44-4462-488b-85e6-8f58024e0f53.jpeg)
9) Enter PIN or provide biometrics.
10) It will fail.
![ohnoez](https://user-images.githubusercontent.com/107238028/173058464-e18bbce0-3417-481a-a5d7-97de533c0856.jpeg)

**Expected behavior**

Wherever the secrets are stored and mapped to a wallet should not use the wallet's friendly name alone to discern where the secret is stored. It should be possible to have an XMR wallet called _widgets_ and also an LTC wallet called _widgets_ if that is what the user prefers. The creation of the second wallet of a different coin should not corrupt the first wallet of the same name. 

**Actual behavior**

If a wallet of any coin is created and given the friendly name of any existing wallet of any other coin, that original wallet will be destroyed.

During testing I wanted to see how the XMR-only monero.com wallet performs on Android and there I came across a cosmetic bug which I filed separately #377.

**Comment by SamsungGalaxyPlayer** [comment]
https://github.com/cake-tech/cake_wallet/pull/391
